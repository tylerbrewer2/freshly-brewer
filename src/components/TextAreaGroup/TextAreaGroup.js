import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// components
import Label from '../Label';
import TextArea from '../TextArea';
import Col from '../Col';

const TextAreaGroupCore = ({
    className,
    labelText,
    value,
    onChange,
    identifier,
    required,
    width
  }) => {
  return (
    <Col width={width}>
      <div className={className}>
        <Label>{labelText}</Label>
        <TextArea
          value={value}
          onChange={onChange}
          identifier={identifier}
        />
      </div>
    </Col>
  )
}

TextAreaGroupCore.propTypes = {
  labelText: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  identifier: PropTypes.string,
  required: PropTypes.bool,
  width: PropTypes.number,
}

TextAreaGroupCore.defaultProps = {
  identifier: null,
  required: false,
  width: 12,
}

const TextAreaGroup = styled(TextAreaGroupCore)`
margin-bottom: 30px;
`

export default TextAreaGroup;