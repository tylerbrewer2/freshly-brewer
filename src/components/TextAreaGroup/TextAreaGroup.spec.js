import React from 'react';
import renderer from 'react-test-renderer';

import TextAreaGroup from './TextAreaGroup';

describe('component structure', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <TextAreaGroup
        labelText="Test label"
        value="Test value"
        identifier="identifier"
        onChange={jest.fn()}
        required
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
