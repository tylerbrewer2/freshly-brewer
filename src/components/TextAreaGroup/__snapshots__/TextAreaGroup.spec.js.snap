exports[`component structure renders correctly 1`] = `
<div
  className="col-xs-12 
      undefined 
      undefined">
  <div
    className="sc-htpNat hvvDZo">
    <label
      className="sc-bdVaJa fxdtZh"
      htmlFor="identifier">
      Test label
    </label>
    <textarea
      className="sc-bwzfXH hQSOcI"
      id="identifier"
      name="identifier"
      onChange={[Function]}
      value="Test value" />
  </div>
</div>
`;
